modules/environment/aws/code-services/main.tf

```diff
@@ -202,3 +202,52 @@ resource "aws_sqs_queue_policy" "code_services_queue_policy" {
 }
 POLICY
 }
+
+resource "aws_iam_role" "sqs_role" {
+  count  = var.enable_aws_code_services ? 1 : 0
+  name = "${var.cluster}_sqs_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${var.account_id}:oidc-provider/${replace(var.openid_connect_provider_url, "https://", "")}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${replace(var.openid_connect_provider_url, "https://", "")}:aud": "sts.amazonaws.com"
```

Comment:
I think it's fine to validate the audience but it's probably more important to validate the subject to ensure that only the service account for this event mapper can assume this role